forwarder.go

```diff
@@ -10,24 +10,44 @@ import (
 	"net/http"
 	"os"
 	"strings"
+	"sync"
 )
 
+const workers = 8
+
 func main() {
 	log.Println("Splunk forwarder: Started")
 	defer log.Println("Splunk forwarder: Stopped")
 
+	forSplunk := make(chan string)
+
+	var wg sync.WaitGroup
+
+	for i := 0; i < workers; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for msg := range forSplunk {
+				postToSplunk(msg)					
+			}
+		}()
+	}
+
 	br := bufio.NewReader(os.Stdin)
 	for {
 		str, err := br.ReadString('\n')
 
 		if err != nil {
 			if err == io.EOF {
+				close(forSplunk)
 				return
 			}
 			log.Fatal(err)
 		}
-		postToSplunk(str)
+		forSplunk <- str
 	}
+
+	wg.Wait()
```

Comment:
@khatton-ft looking at other examples like https://golang.org/pkg/os/signal/#Stop the statement forSplunk <- str blocks until new string is received from statement str, err := br.ReadString('\n').

Based on this your comment about wg.Wait() never been called is true because for loop never exits. 

I've run tests with and without the wg.Wait() on line 49. Both works fine. 

Options here are to remove the line or leave it in. 

Which one you prefer?